src/effects/Reverse.cpp

```diff
@@ -182,12 +182,12 @@ bool EffectReverse::ProcessOneWave(int count, WaveTrack * track, sampleCount sta
             double offsetStartTime = track->LongSamplesToTime(clipOffsetStart);
             if(i+1 < clipArray.size()) // update currentEnd if there is a clip to process next
             {
-               auto nextClipStart = clipArray[i+1]->GetStartSample();
+               auto nextClipStart = clipArray[i+1]->GetPlayStartSample();
                currentEnd = currentEnd - (clipEnd - clipStart) - (nextClipStart - clipEnd);
             }
 
             revClips.push_back(track->RemoveAndReturnClip(clip)); // detach the clip from track
-            revClips.back()->SetOffset(track->LongSamplesToTime(track->TimeToLongSamples(offsetStartTime))); // align time to a sample and set offset
+            revClips.back()->SetPlayStartTime(track->LongSamplesToTime(track->TimeToLongSamples(offsetStartTime))); // align time to a sample and set offset
```

Comment:
As I review the necessary changes in effects, this occurs to me.  What is expected if effects are applied to clips with hidden parts?  If you do nothing else, then the hidden parts remain unchanged.  Perhaps that is expected behavior.  Or perhaps in the case of the Reverse effect, it would be surprising behavior.  I’m not sure whether you should do anything extra.

This is one of the few places where there was a call to SetOffset.  Other effects did not use it.